### PR TITLE
[WIP] Python Implementation of Space API

### PR DESCRIPTION
**Pull request by AliabbasMerchant**
I have tried to mirror the JS API, but have made the variable names pythonic.

Also, for cases where we do not have an equivalent gRPC method, I have raised a `NotImplementedError`

### CODE REVIEW

**Comment by YourTechBud** [comment]
The logic seam to be right according to me. Can you remove the .idea/* files?

**Comment by YourTechBud** [comment]
By when will you prepare the docs? And how are you doing it?

**Comment by AliabbasMerchant** [comment]
I will make python docstrings.
Then, mostly use sphinx etc to make them into a webpage sort of.

I am starting docs now.

**Comment by YourTechBud** [comment]
you can refer to the js docs [here](https://github.com/spaceuptech/space-api-js/wiki). Also, we have updated the main [docs](https://spaceuptech.com/docs). We have to add the python snippets there as well.

File: space_api/mongo/update.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    :param token: (str) The (optional) JWT Token
+    """
+
+    def __init__(self, project_id: str, collection: str, url: str, token: Optional[str] = None):
+        self.project_id = project_id
+        self.collection = collection
+        self.url = url
+        self.db_type = "mongo"
+        self.token = token
+        self.params = {'find': {}, 'update': {'$set': {}}}
```
Comment (YourTechBud) [inline space_api/mongo/update.py]: Let update be an empty map. Don't add the `$set` field in the constructor.

**Pull request merged**